Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
Traceback (most recent call last):
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 298, in <module>
    main()
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 280, in main
    prepare_experiment(
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 137, in prepare_experiment
    result = run_experiment(
             ^^^^^^^^^^^^^^^
  File "/home/vincent.opitz/master-thesis/GAIA/src/run.py", line 80, in run_experiment
    cluster.save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/simulation_cluster.py", line 93, in save_results
    super().save_results(
  File "/home/vincent.opitz/master-thesis/GAIA/src/cluster/base_cluster.py", line 182, in save_results
    df.to_csv(details_filename, index=False)
  File "/usr/local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/generic.py", line 3967, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/usr/local/lib/python3.12/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 749, in get_handle
    check_parent_directory(str(handle))
  File "/usr/local/lib/python3.12/site-packages/pandas/io/common.py", line 616, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'results/simulation/different_lengths'
slurmstepd-cx17: error: *** JOB 1207849 ON cx17 CANCELLED AT 2024-09-29T19:01:33 ***
